cluster/shard_writer.go

```diff
@@ -43,7 +43,9 @@ func (w *ShardWriter) WriteShard(shardID, ownerID uint64, points []tsdb.Point) e
 	if !ok {
 		panic("wrong connection type")
 	}
-	defer conn.Close() // return to pool
+	defer func(conn net.Conn) {
```

Comment:
Can you explain why? Isn't the `conn` object the same in either case? What am I missing?